### ISSUE TITLE
Unable to send push for Android with Meteor 1.3.2.4

### ISSUE THREAD

**Issue by chrisco23** [raix/push#214]
Hi,

Sorry I left this as a comment on Issue #153 but this morning it occurs to me that in light of the different error situation, I should probably have created a separate issue.  So I'm doing that here.  Here was my comment:

Glad I found this thread as it seems to get me closer.

After running 'meteor add cordova:cordova-plugin-device@1.1.1', I get

`I20160511-00:28:02.938(-7) (android:http://localhost:12488/706a87c28afe2f2833e09f1faae76451d1e9bdb6.js:179) Push.Token:
I20160511-00:28:02.943(-7) (android:http://localhost:12488/706a87c28afe2f2833e09f1faae76451d1e9bdb6.js:223) Token: {"gcm":"APA91bECNHQBuGl1W0DdANTBcX3bafsxwABndUhKCl3KZGT0xWpoLOjL95GjdgkUGxPR-k9J7kFdncH3V7InSaIEOB6X6BQrJrQssM8ev5jTn7nSqidjazVMxJqvZfm033PaWO8b0LEl"}
I20160511-00:28:02.944(-7) (android:http://localhost:12488/706a87c28afe2f2833e09f1faae76451d1e9bdb6.js:179) Got token:
I20160511-00:28:16.770(-7) (android:http://localhost:12488/706a87c28afe2f2833e09f1faae76451d1e9bdb6.js:223) sending notification
I20160511-00:28:18.696(-7) (android:http://localhost:12488/706a87c28afe2f2833e09f1faae76451d1e9bdb6.js:179) Push.Notification:
I20160511-00:28:18.697(-7) (android:http://localhost:12488/706a87c28afe2f2833e09f1faae76451d1e9bdb6.js:179) Push.Message: Got message while app is open:`

Now, I am still using some Meteor.startup() code from when I first successfully used raix:push, last July(!).  I have this code from back then still:

`Meteor.startup(function () {
  console.log('inside notifications.js');
  Push.allow({
    send: function(userId, notification) {
      // Allow all users to send to everybody - For test only!
      return true;
    }
  });

  Push.debug = true;

  /* do something with push token*/
  Push.addListener('token', function(token) {
    console.log('Token: ' + JSON.stringify(token));
  });

  Push.addListener('notification', function(notification) {
    console.log('notification: ' + JSON.stringify(notification))
  });

  Push.addListener('startup', function(notification) {
    console.log('Routing Push: ' + JSON.stringify(notification));
    Router.go('warbleShow', {_id: notification.payload.warbleId});
  });
});
`

In light of the messages I mentioned in that first bit of output, I tried closing the application and sending the test notification from localhost:3000 (is that even supposed to work??), and I got:

`I20160511-00:30:40.853(-7)? Push: Send message "Hello" via query {}
I20160511-00:30:40.855(-7)? Push: Sent message "Hello" to 0 ios apps 0 android apps
I20160511-00:30:40.857(-7)? Push, GUIDE: The "Push.appCollection" is empty - No clients have registred on the server yet...
`

**Comment by chrisco23** [comment]
In case this is helpful, I found this in Google Play's GCM Diagnostics:

MESSAGE 0:1462988289872828%F1CA786B0034AF1A

Current Status	Acknowledged
Collapse Key	push
TTL	2,419,200
TIME	CURRENT STATUS
May 10, 11:57 PM
Device connected
May 11, 10:38 AM
Accepted
May 11, 10:38 AM
Delivered
May 11, 11:04 AM
Acknowledged

.. that was for my most recent token..  the previous page shows:

May 11, 10:38 AM
0:1462988289872828%f1ca786b0034af1a
push
2,419,200
Acknowledged
May 11, 10:29 AM
0:1462987749916129%f1ca786b0034af1a
push
n/a
Acknowledged
May 11, 10:23 AM
0:1462987389723346%f1ca786b0034af1a
push
n/a
Collapsed
May 11, 10:22 AM
0:1462987344907114%f1ca786b0034af1a
push
n/a
Collapsed
May 11, 12:28 AM
0:1462951698354190%f1ca786b0034af1a
push
n/a
Acknowledged

**Comment by wanchopeblanco** [comment]
Im having a similar where android devices wont register a token but ios devices do register their tokens.

Im using Meteor 1.3 and config.configure method on client and server.

**Comment by etyp** [comment]
Having the same issue here

**Comment by MaxShv** [comment]
+1

I20160630-11:29:50.679(-4)? Push: Sent message "test" to 0 ios apps 1 android apps
I20160630-11:29:50.784(-4)? Unauthorized
I20160630-11:29:50.787(-4)? ANDROID: Result of sender is null

**Comment by mvgalle** [comment]
Same issue here. Devices not registering.

**Comment by HedCET** [comment]
+1, Devices not registering

**Comment by crazybaud** [comment]
Same issue on our side, and it was due to bad GCM server key, `gcm.apiKey`in our `config.push.json`. We changed it here : https://console.cloud.google.com/apis/credentials

**Comment by jensljungblad** [comment]
I'm getting the same thing on Meteor 1.3.4.4:

I20160630-11:29:50.679(-4)? Push: Sent message "test" to 0 ios apps 1 android apps
I20160630-11:29:50.784(-4)? Unauthorized
I20160630-11:29:50.787(-4)? ANDROID: Result of sender is null

@MaxShv did you find a workaround?

@crazybaud I tried regenerating the api key a bunch of times but no luck.. What version of Meteor are you on and how have you configured push?

**Comment by crazybaud** [comment]
Our install : 
- With Meteor 1.3.4.1
- `meteor add raix:push@3.0.3-rc.5`
- We don't use anymore `config.push.json` file.

Client side : 
```
// Configure our notification
Push.Configure({
	"gcm": {
		"projectNumber": "[12 digits]"
	},
	"bagde": true,
	"sound": true,
	"alert": true,
	"vibrate": true
});
```

Server side : 
```
// Configure our notification
Push.debug = true;
Push.Configure({
  "apn": {
    "passphrase": "[yours]",
    "keyData": Assets.getText('apnDevKey.pem'),
    "certData": Assets.getText('apnDevCert.pem')
  },
  "gcm": {
    "apiKey": "[yours]",
    "projectNumber": "[same digits]"
  },
  "production": true
});
```

`mobile-config.js` : 
```
App.configurePlugin('phonegap-plugin-push', {
	SENDER_ID: '[same digits]'
});
```

hope it helps :)

**Comment by sschuchlenz** [comment]
Same here and no solution posted in the different issues seems to help. Currently we are using the rc.5 branch with config similar to the one @crazybaud mentioned (only differences are that in one of the other issues it was stated that the project number has to be integer so we tried with and without quotes and we left out APN config due to the fact that we do not support iOS yet) and we've already tried to change the API key in Google's console.

We've also tried with installing `cordova-plugin-device@1.1.1` and different other approaches but to no avail.

**Comment by raix** [comment]
I have to give 1.4.x a go - haven't had the time to follow their development as closely as I used to - I've cleaned up branches and bumped version to `3.0.3-rc.7` - the `device undefined` issue was solved a while ago.

*Mind I still need to do QA on `3.0.3-rc.7` making sure things work - it's time consuming because I literally have to create an app for both ios and android + test cases*

**Comment by sschuchlenz** [comment]
@raix thanks for looking into it - we've decided to stick with OneSignal and cordova plugin integration for now but once yours is ready for 1.4, we'll give it a test drive

**Comment by cunneen** [comment]
@chrisco23 See my [comment on #240](https://github.com/raix/push/issues/240#issuecomment-248540342) regarding making sure push notification entitlements are enabled.

**Comment by crazybaud** [comment]
@jensljungblad we upgraded to 1.4.2-beta4 and it is not working anymore on Android (Intercom notification system is functional).

**Comment by jensljungblad** [comment]
@raix I still can't get Android to work on 1.3.4.4 even with 3.0.3-rc.7. 

My config:

```js
// Client
Push.Configure({
  android: {
    senderID: '12345',
    badge: true,
    sound: true,
    alert: true,
    vibrate: true
  }
});

// Server
Push.Configure({
  gcm: {
    apiKey: 'abcde',
    projectNumber: '12345',
  }
});

// mobile-config.js
App.configurePlugin('phonegap-plugin-push', {
  SENDER_ID: '12345'
});
```

After which I get the following error when pushing a message:

```
Push: Sent message "Foobar" to 0 ios apps 1 android apps
Unauthorized
ANDROID: Result of sender is null
```

**Comment by cunneen** [comment]
@jensljungblad check that your IP address has been added to your Google console. Also try using the latest devel branch of raix:push (which I published as appworkshop:push@3.0.3-rc.8 just to make it easier): 

```
meteor remove raix:push
meteor add appworkshop:push@3.0.3-rc.8
```

Note that I won't be maintaining that version, you should switch back to raix:push as soon as the next version is published.

**Comment by aqib1604** [comment]
@jensljungblad I have same issue with android Did you found any workaround?

**Comment by cunneen** [comment]
@aqib1604 did you see my message above yours?

**Comment by aqib1604** [comment]
When I run by adding your package I am getting following error.

Same error is coming with raix:push 3.0.3 rc-7
```
While running Cordova app for platform Android with options --device:
   Error: Command failed: /Users/Trainee6/Desktop/Backup/webapp_17sep_Backup/.meteor/local/cordova-build/platforms/android/cordova/run --device
   FAILURE: Build failed with an exception.
   
   * What went wrong:
   A problem occurred configuring root project 'android'.
   > Could not resolve all dependencies for configuration ':_armv7DebugCompile'.
   > Could not find any version that matches com.google.android.gms:play-services-gcm:9.0.2+.
   Searched in the following locations:
   https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/maven-metadata.xml
   https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/
   https://download.01.org/crosswalk/releases/crosswalk/android/maven2/com/google/android/gms/play-services-gcm/maven-metadata.xml
   https://download.01.org/crosswalk/releases/crosswalk/android/maven2/com/google/android/gms/play-services-gcm/
   file:/Users/Trainee6/Library/Android/sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/maven-metadata.xml
   file:/Users/Trainee6/Library/Android/sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/
   file:/Users/Trainee6/Library/Android/sdk/extras/google/m2repository/com/google/android/gms/play-services-gcm/maven-metadata.xml
   Required by:
   :android:unspecified
   
   * Try:
   Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
   Error code 1 for command: /Users/Trainee6/Desktop/Backup/webapp_17sep_Backup/.meteor/local/cordova-build/platforms/android/gradlew with args:
   cdvBuildDebug,-b,/Users/Trainee6/Desktop/Backup/webapp_17sep_Backup/.meteor/local/cordova-build/platforms/android/build.gradle,-PcdvBuildArch=arm,-Dorg.gradle.daemon=true,-Pandroid.useDeprecatedNdk=true undefined
   at ChildProcess.exitCallback (/tools/utils/processes.js:151:23)
   at ChildProcess.emit (events.js:98:17)
   at Process.ChildProcess._handle.onexit (child_process.js:820:12)
```

**Comment by aqib1604** [comment]
@cunneen @jensljungblad  I have made it working through `meteor add appworkshop:push@3.0.3-rc.8` and [https://github.com/raix/push/blob/master/docs/ANDROID.md](https://github.com/raix/push/blob/master/docs/ANDROID.md).

Thanks for the help @cunneen

**Comment by jensljungblad** [comment]
Got it working by following the updated Android instructions and switching to FCM. Thanks!
